specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/Import.tsx

```diff
@@ -0,0 +1,368 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { usePromise } from '../../hooks/useAsyncState';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { RA } from '../../utils/types';
+import { Container } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import type { UiFormatter } from '../Forms/uiFormatters';
+import { Dialog, LoadingScreen } from '../Molecules/Dialog';
+import { FilePicker } from '../Molecules/FilePicker';
+import { TableIcon } from '../Molecules/TableIcon';
+import { NotFoundView } from '../Router/NotFoundView';
+import { fetchAttachmentResourceId } from './fetchAttachmentResource';
+import { staticAttachmentImportPaths } from './importPaths';
+import { RenameAttachmentDataSetDialog } from './RenameDataSet';
+import { SafeRollbackAttachmentsNew } from './Rollback';
+import { SelectUploadPath } from './SelectUploadPath';
+import type {
+  AttachmentDataSetResource,
+  FetchedDataSet,
+  PartialUploadableFileSpec,
+  SavedAttachmentDataSetResource,
+  UnBoundFile,
+} from './types';
+import { SafeUploadAttachmentsNew } from './Upload';
+import { useEagerDataSet } from './useEagerDataset';
+import {
+  matchSelectedFiles,
+  reconstructDeletingAttachment,
+  reconstructUploadingAttachmentSpec,
+  resolveFileNames,
+} from './utils';
+import { AttachmentsValidationDialog } from './ValidationDialog';
+import { ViewAttachmentFiles } from './ViewAttachmentFiles';
+
+export type AttachmentUploadSpec = {
+  readonly staticPathKey: keyof typeof staticAttachmentImportPaths;
+  readonly formatQueryResults: (
+    value: number | string | null | undefined
+  ) => string | undefined;
+};
+export type PartialAttachmentUploadSpec = {
+  readonly fieldFormatter?: UiFormatter;
+} & (AttachmentUploadSpec | { readonly staticPathKey: undefined });
+
+type AttachmentDataSet =
+  | AttachmentDataSetResource
+  | SavedAttachmentDataSetResource;
+export type EagerDataSet = AttachmentDataSet & {
+  readonly needsSaved: boolean;
+  readonly save: boolean;
+};
+
+const newAttachmentDataSetResource: AttachmentDataSetResource = {
+  name: attachmentsText.newAttachmentDataset(),
```

Comment:
(optional) consider adding a date in the name so that people can create data sets without having to rename them at all

Reply:
and consider deduplicating the names to force all names to be unique, just like we are forcing right now for workbench data sets

Reply:
Add timestamp like the WorkBench:
`New Bulk Attachment Import – Fri Jul 28 2023`

Reply:
add a (2) or (3) after when a duplicate occurs

Reply:
Fixed. Uses the same component as workbench now